WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -0,0 +1,641 @@
+            case POPULAR:
+                adapter = (PluginBrowserAdapter) mPopularPluginsRecycler.getAdapter();
+                cardView = findViewById(R.id.popular_plugins_cardview);
+                break;
+            case NEW:
+                adapter = (PluginBrowserAdapter) mNewPluginsRecycler.getAdapter();
+                cardView = findViewById(R.id.new_plugins_cardview);
+                break;
+            default:
+                adapter = (PluginBrowserAdapter) mSitePluginsRecycler.getAdapter();
+                cardView = findViewById(R.id.installed_plugins_cardview);
+                break;
+        }
+
+        List<?> plugins = getPlugins(pluginType);
+        adapter.setPlugins(plugins);
+
+        int newVisibility = plugins.size() > 0 ? View.VISIBLE : View.GONE;
+        int oldVisibility = cardView.getVisibility();
+        if (newVisibility == View.VISIBLE && oldVisibility != View.VISIBLE) {
+            AniUtils.fadeIn(cardView, AniUtils.Duration.MEDIUM);
+        } else if (newVisibility != View.VISIBLE && oldVisibility == View.VISIBLE) {
+            AniUtils.fadeOut(cardView, AniUtils.Duration.MEDIUM);
+        }
+    }
+
+    private List<?> getPlugins(@NonNull PluginListType pluginType) {
+        switch (pluginType) {
+            case POPULAR:
+                return mPluginStore.getPluginDirectory(PluginDirectoryType.POPULAR);
+            case NEW:
+                return mPluginStore.getPluginDirectory(PluginDirectoryType.NEW);
+            default:
+                return mPluginStore.getSitePlugins(mSite);
+        }
+    }
+
+    private SitePluginModel getSitePluginFromSlug(@Nullable String slug) {
+        if (slug != null) {
+            for (SitePluginModel plugin : mSitePlugins) {
+                if (slug.equals(plugin.getSlug())) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSitePluginsFetched(PluginStore.OnSitePluginsFetched event) {
+        if (isFinishing()) {
+            return;
+        }
+        showProgress(false);
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching site plugins with type: " + event.error.type);
+            ToastUtils.showToast(this, R.string.plugin_fetch_error);
+            return;
+        }
+        refreshPlugins(PluginListType.SITE);
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onWPOrgPluginFetched(PluginStore.OnWPOrgPluginFetched event) {
+        if (isFinishing()) {
+            return;
+        }
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching the wporg plugin with type: " + event.error.type);
+            return;
+        }
+        if (!TextUtils.isEmpty(event.pluginSlug)) {
+            refreshPluginWithSlug(event.pluginSlug);
```

Comment:
Some plugins don't exist in the directory and FluxC doesn't return an error for them which results in an infinite loop here. I've opened https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/704 to fix it.